source/slang/slang-lower-to-ir.cpp

```diff
@@ -6014,13 +6017,25 @@ struct StmtLoweringVisitor : StmtVisitor<StmtLoweringVisitor>
         {
             if (context->returnDestination.flavor != LoweredValInfo::Flavor::None)
             {
-                // If this function should return via a __ref parameter, do that and return void.
-                lowerRValueExprWithDestination(context, context->returnDestination, expr);
-                getBuilder()->emitReturn();
-                return;
+                if (constructorDecl)
```

Comment:
In this branch we should not do the `if(constructorDecl)`, Noncopyable types's constructor should be handled the same as ordinary functions as in 6032-6035.

Reply:
thanks for clarifying this. i got confused with the different scenarios, without trying an actual test case for each.